### PR TITLE
analysis tool writes object to file

### PR DESCRIPTION
**Pull request by Raulp8**

### CODE REVIEW

File: analysis/analysis.go [hunk 1]
```diff
@@ -27,15 +27,25 @@ import (
...
 	}
-	generateAnalysis(os.Args[1])
+	dir := os.Args[1]
+	analysis := generateAnalysis(dir)
+	writeAnalysis(dir, analysis)
+}
+
+func writeAnalysis(analysisName string, analysis *incompatibility.ApiSetIncompatibility) {
+	pbMessage, msgErr := utils.ProtoTextBytes(analysis)
+	exitIfError(msgErr)
```
Comment (skeid21) [inline analysis/analysis.go:44]: This isn't very idiomatic of go.  The more common approach is to return an error.  At the top-level main function, you would translate that into an exit. 

This does slightly obfuscate that the application could exit for example when this method is called from main. 
I realize gnostic uses this pattern.

Thoughts?

File: analysis/analysis.go [hunk 2]
```diff
@@ -27,15 +27,25 @@ import (
...
 
 // main function for aggreation tool
 func main() {
 	if len(os.Args) != 2 {
 		exitIfError(errors.New("argument should be a path to a directory"))
 	}
-	generateAnalysis(os.Args[1])
+	dir := os.Args[1]
+	analysis := generateAnalysis(dir)
+	writeAnalysis(dir, analysis)
```
Comment (skeid21) [inline analysis/analysis.go:39]: Should we exit with 0 on success?

File: analysis/analysis.go (hunk 1 above)
Comment (Raulp8) [reply analysis/analysis.go]: error now handled at main

File: analysis/analysis.go (hunk 2 above)
Comment (Raulp8) [reply analysis/analysis.go]: now exits with 0 at end of main indicating success

**Review by AlanPeters — APPROVED** [review]
LGTM after addressing Shawn's comments

**Review by skeid21 — APPROVED** [review]

**Pull request merged**